metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java

```diff
@@ -56,18 +60,19 @@
 
     private WireMockServer wireMockServer;
 
-    @ParameterizedTest
+    @ParameterizedTest(name = "#{index} - {0}")
     @MethodSource({
-            "provideMetadataForTags",
-            "provideMetadataForAbstract",
-            "provideMetadataForTitle",
-            "provideMetadataForDate",
-            "provideMetadataForContributors",
-            "provideMetadataForIdentifier",
-            "provideMetadataForLanguage"
+        "provideMetadataForTags",
+        "provideMetadataForAbstract",
+        "provideMetadataForTitle",
+        "provideMetadataForDate",
+        "provideMetadataForContributors",
+        "provideMetadataForIdentifier",
+        "provideMetadataForLanguage"
     })
-    public void getCreatePublicationParsesHtmlAndReturnsMetadata(String html, CreatePublicationRequest expectedRequest)
-            throws IOException {
+    public void getCreatePublicationParsesHtmlAndReturnsMetadata(String testDescription, String html,
```

Comment:
testDescription is not being used.

Reply:
It actually is; by the mapper. Without the correct number of arguments in the right position it gets confused. If there's a better way of solving this issue I'd welcome your suggestion.

Reply:
i think the problem lies in somehow the parametrized test. It has become given a "special case input" assert that the parser parses it. Basically the definition of the test has become a parameter and that is confusing.

I think the tests have become a bit too complicated to read and grouped in an unexpected way. It would be more clear if we had tests that were 
1. this field maps to this field
2. The parser can handle upper-cased field values
3. The parser can handle lower-cased field values

Instead we have 
1. check lowercase abstract and lowercase title
2. check uppercase title and citation title.

My suggestion is : 
```
  @Test
    public  void dcTitleMapsToMainTitle() throws IOException {
        String title = "Title";
        CreatePublicationRequest request = createRequestWithTitle(title);
        Map<String, String> titleMap = Map.of(DC_TITLE_LOWERCASE, title);
        String html = createHtml(titleMap);
        assertThatParsedHtmlContainsExpectedMetadata(html, request);
    }

    @Test
    public  void lowerCasedFieldsMapCorrectly() throws IOException {
        String title = "Title";
        CreatePublicationRequest expectedRequest = createRequestWithLowerCaseFields(title);
        Map<String, String> lowerCaseFields = lowerCaseFields();
        String html = createHtml(lowerCaseFields);
        assertThatParsedHtmlContainsExpectedMetadata(html, expectedRequest);
    }

    private void assertThatParsedHtmlContainsExpectedMetadata(String html, CreatePublicationRequest expectedRequest)
        throws IOException {
        String filename = "article.html";
        URI uri = prepareWebServerAndReturnUriToMetadata(filename, html);
        MetadataService metadataService = new MetadataService();
        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);

        assertThat(request.isPresent(), is(true));
        CreatePublicationRequest actual = request.get();
        actual.setContext(null);

        assertThat(actual, is(equalTo(expectedRequest)));
        
        stopServer();
    }

    @AfterEach
    public void stopServer() {
        wireMockServer.stop();
    }

```

Reply:
We do not have to fix this now though

Reply:
Individual tests (conventional JUNIT use?) were brought up as a solution. I think we decided against it thinking there would be too much repeat of boilerplate, setup and teardown. The thinking being that this is more compact. However, I agree with you. It's not ideal. I feel like parametrized tests are nice, but should be used more to send varied, but mostly similar parameters into a single test, not as a way of handling setup and grouping of tests like we currently are.

Especially as we have more specific test cases then the ones you summarize, which is part of what prompted this change. This is a marked improvement though, we can revisit this again.